doc/REFERENCE_COUNTER.md

```diff
@@ -30,4 +30,17 @@ Essentially the only time there is  total blockstore lock preventing any calls i
 
 The biggest performance sink hole in this is the garbage collection process, not the reference counting process. In and of itself garbage collection is pretty quick with our reference counter as we just need to read key->value pairs without any graph traversal, however  depending on how much data you are garbage collecting at any one momment, the biggest impact on the performance will be your disk speeds.
 
-Thankfully due to our multi-datastore configuration capabilities, if you are in an environment where performance is of extreme importance, you can place this metadata store on an NVMe SSD, or regular SSD drive and this will mitigate a large portion of the garbage collection overhead.
\ No newline at end of file
+Thankfully due to our multi-datastore configuration capabilities, if you are in an environment where performance is of extreme importance, you can place this metadata store on an NVMe SSD, or regular SSD drive and this will mitigate a large portion of the garbage collection overhead.
+
+# Migrating From Queue Based Counter To Queueless Counter
```

Comment:
I think there should be a specific doc for upgrading that is referenced from the beginning of the main readme instead of hidden in here.

This should only describe the current implementation with a passing reference to historical and upgrading information.

Reply:
good point i fixed this